zebra-state/src/service/read/find.rs

```diff
@@ -53,7 +91,52 @@ where
         || db.contains_hash(hash)
 }
 
-/// Find the first hash that's in the peer's `known_blocks` and the chain.
+/// Create a block locator from `chain` and `db`.
+pub fn block_locator<C>(chain: Option<C>, db: &ZebraDb) -> Option<Vec<block::Hash>>
+where
+    C: AsRef<Chain>,
+{
+    let chain = chain.as_ref();
+
+    let tip_height = tip_height(chain, db)?;
+
+    let heights = block_locator_heights(tip_height);
+    let mut hashes = Vec::with_capacity(heights.len());
+
+    for height in heights {
+        if let Some(hash) = hash_by_height(chain, db, height) {
+            hashes.push(hash);
+        }
+    }
+
+    Some(hashes)
+}
+
+/// Get the heights of the blocks for constructing a block_locator list.
```

Comment:
The following suggestion is unrelated to the PR, but I think it's worth having it in the code since it took me a while to figure out how the function assembles the heights.

```suggestion
/// Gets the heights of the blocks for constructing a block_locator list.
///
/// The initial height in the returned `vec` is the tip height. The initial distance
/// between heights is 1. This distance increases exponentially with base 2 
/// between each subsequent height. The distance between the first and last
/// height is at most [`constants::MAX_BLOCK_REORG_HEIGHT`], and the
/// number of returned heights is approximately `log_2` of this value.
```

Reply:
Thanks! Some of these things are implementation details, so I might put them inside the function in comments. But the rest I'll put in the function docs.

Reply:
What do you think of my changes in f1f2230f9b ?

Reply:
It's probably too much nit-picking at this point, but I think the expression _"exponentially decreasing"_ is not the right description in this case.

An exponentially decreasing quantity decreases fast when it's large and slow when it's low. I'd depict such a decrease by plotting the function e^-x:
![image](https://user-images.githubusercontent.com/8863307/189976547-1379b68c-ccb2-4890-8613-40d121218ead.png)


However, in our case, we have the "inverse" type of decreasing --- slow when the value is large, and fast as the value gets lower. I'd depict such a decrease by plotting the function -e^x:
![image](https://user-images.githubusercontent.com/8863307/189976661-812cb4d5-c532-47fb-8dd2-f1933168e25f.png)
 I was trying to find a name for this type of decrease, but nothing came up. I would still rather avoid the expression _"exponentially decreasing"_ here.

Reply:
It's probably too much nit-picking at this point, but I think the expression _"exponentially decreasing"_ is not the right description in this case.

An exponentially decreasing quantity decreases fast when it's large and slow when it's low. I'd depict such a decrease by plotting the function e^-x:
![image](https://user-images.githubusercontent.com/8863307/189976547-1379b68c-ccb2-4890-8613-40d121218ead.png)


However, in our case, we have the "inverse" type of decreasing --- slow when the value is large, and fast as the value gets lower. I'd depict such a decrease by plotting the function -e^x:
![image](https://user-images.githubusercontent.com/8863307/189976661-812cb4d5-c532-47fb-8dd2-f1933168e25f.png)
 I was trying to find a name for this type of decrease, but nothing came up. I would still rather avoid the expression _"exponentially decreasing"_ here.

Reply:
I fixed this in 9efa75640.

Since this change isn't blocking, and isn't related to this PR, let's open another PR for any further changes?

Reply:
Definitely. Sorry for the mess.

It looks like this type of decrease is not particularly important for humanity, so there was no need to coin a term for it and rigorously define it. Maybe it's a good opportunity for us now? Inverse exponential decay?